language/translation/translation_training_data_tsv_generator.ipynb

```diff
@@ -0,0 +1,786 @@
+    "                                    tsv_f.write(\n",
+    "                                        \" \".join(src_row_data)\n",
+    "                                        + \"\\t\"\n",
+    "                                        + \" \".join(ref_row_data)\n",
+    "                                    )\n",
+    "                                    tsv_f.write(\"\\n\")\n",
+    "                        except Exceptio as e:\n",
+    "                            print(e)\n",
+    "                    else:\n",
+    "                        try:\n",
+    "                            mismatched_block[src_block.text] = ref_block\n",
+    "                        except:\n",
+    "                            mismatched_block[src_block] = ref_block.text\n",
+    "                        break\n",
+    "\n",
+    "            tsv_f.close()\n",
+    "        print(f\"Generated TSV stored at {tsv_file}\")\n",
+    "        return mismatched_block, more_than_200_words\n",
+    "    except Exception as e:\n",
+    "        print(e)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "id": "de7fc4fe-50dd-439a-931d-aed77ded7053",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Generated TSV stored at /home/jupyter/src/your_tsv_file_name.tsv\n"
+     ]
+    }
+   ],
+   "source": [
+    "mismatched_block, more_than_200_words = make_tsv(source_ref_dictionary, tsv_file_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 73,
+   "id": "040d325e-9ef8-47fd-9bf7-b11c68f208d5",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{}"
+      ]
+     },
+     "execution_count": 73,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "mismatched_block"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 74,
+   "id": "51ceb169-ecb0-401d-ab6d-07aa1ebd3b77",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{}"
+      ]
+     },
+     "execution_count": 74,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "more_than_200_words"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8c4bb7f5-4df1-445d-b97f-b026ed98ad92",
+   "metadata": {},
+   "source": [
+    "## Custom model training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 85,
+   "id": "a89cbbda-8996-447e-abec-96de3c8bc71c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def generate_access_token() -> Credentials:\n",
+    "    \"\"\"Generates access token to call translate APIs.\"\"\"\n",
+    "    creds, project = google.auth.default()\n",
+    "\n",
+    "    auth_req = google.auth.transport.requests.Request()\n",
+    "    creds.refresh(auth_req)\n",
+    "    return creds.token\n",
```

Comment:
This shouldn't be needed if you use the Client Library

Reply:
Please see my comment  here:https://github.com/GoogleCloudPlatform/generative-ai/pull/880#discussion_r1685957623